src/components/app.jsx

```diff
@@ -1,25 +1,96 @@
 import React from 'react';
-import PropTypes from 'prop-types';
 import LocationDetails from './location-details';
 import ForecastSummaries from './forecast-summaries';
+import ForecastDetails from './forecast-details';
+import SearchForm from './search-form';
 import '../styles/app.scss';
+import axios from 'axios';
 
-const App = props => (
-  <div className="forecast">
-    <LocationDetails
-      city={props.location.city}
-      country={props.location.country}
-    />
-    <ForecastSummaries forecasts={props.forecasts} />
-  </div>
-);
-
-App.propTypes = {
-  location: PropTypes.shape({
-    city: PropTypes.string,
-    country: PropTypes.string,
-  }).isRequired,
-  forecasts: PropTypes.array.isRequired,
-};
+const URL = 'https://mcr-codes-weather.herokuapp.com/forecast';
+
+class App extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      selectedDate: 0,
+      forecasts: [],
+      location: {
+        city: '',
+        country: '',
+      },
+      error: false,
+      errorLocation: '',
+    };
+  }
+
+  componentDidMount() {
+    this.handleLocationSelect(this.state.location.city);
+  }
+
+  handleForecastSelect = (date) => {
+    this.setState({
+      selectedDate: date,
+    });
+  };
+
+  handleLocationSelect = (location) => {
+    axios.get(`${URL}?city=${location}`)
+      .then(response => {
+        const data = response.data;
+        this.setState({
+          selectedDate: data.forecasts[0].date,
+          forecasts: data.forecasts,
+          location: {
+            city: data.location.city,
+            country: data.location.country,
+          },
+          errorLocation: '',
+          error: false,
+        });
+      })
+      .catch(() => {
```

Comment:
Good work getting to the error handling. You might like to try using the error the api sends back to decide what information is displayed to the user. 

```
.catch(error => {
  this.setState({
    errorMessage: error.message,
  })
})
```

Reply:
You would also need to think about how you are displaying this information to the user. The quick and dirty solution is to use a `window.alert()` but people tend to find those annoying. A more polished way is to have a `<messageBox />` component with conditional rendering. eg:

```
  if (this.state.error) <messageBox message={this.state.error.message} />
```